packages/lit-dev-content/site/docs/templates/expressions.md

```diff
@@ -207,23 +218,39 @@ If the expression makes up the entire attribute value, you can leave off the quo
 html`<img src="/images/${this.image}">`;
 ```
 
-### Setting attributes if data is defined { #ifDefined }
+### Removing an attribute under certain conditions { #removing-attribute }
```

Comment:
I think it's not great that talking about removing at attribute is here, but the general talk about removing child expressions is either in the conditionals page or under a type-oriented hierarchy above. I think I'd move most of the discussion about rendering nothing in child expressions to a section outside of the type-based organization, that's analogous to this section on attributes, then link to conditionals from that section. And keep the part organized by type just organized by type.

Reply:
So, the type hierarchy is specifically about the types of values for child expressions (because there are so many and there's some complexity). It makes sense to me that the information about removing (or clearing) child expressions is in the child expression section and the information about removing attributes is in the attribute expression section.

I think the only really confusing thing here is that we treat attributes and boolean attributes as separate things. And as a side effect, we end up describing removing an attribute using `nothing` (arguably the advanced use case) before we describe boolean attribute expressions (arguably the simple use case). 

We could make that a little more logical by making the "boolean attribute expressions" section a child of the "Attribute expressions" section, and moving this "Removing an attribute..." section _after_ the boolean attr section.  It makes sense from a user point of view ("I want to ... X") even if it slightly less "correct" in some sense.

Reply:
Or we could just add a "(see below)" parenthetical to the boolean attribute syntax callout; doesn't seem like the end of the world to have a use-before-define in this case, but I think what you said works too.

At this point, both `ifDefined` and arguably boolean attribute `?attr=` syntax are both vestiges of not initially having a sentinel to remove attributes; if we had that, I'm not sure we'd have either (maybe arguable).

As for placement, agree with Arthur that pulling it out into a "rendering nothing" section that covers both child and attr expressions isn't clearly better than describing `nothing`'s behavior per-expression type, since it does do different things per type that needs to be explained.

Reply:
I'm not suggesting a "rendering noting" section that covers both attributes and children. I'm suggesting that both child expressions and attribute expressions sections had "rendering nothing" subsections, because one right now is here and the other is in the conditionals page.

Reply:
Attempted to address feedback, ptal.